### PR TITLE
JBPM-6057: Stunner - fail validation if process can't be loaded after reload

### PR DESCRIPTION
**Pull request by alepintus**
Hi @romartin, @hasys 
Added popup message to prevent deletion of a Process Variable still bound to a User Task or a BusinssRule Task.

Could you please take a look?

Thanks!

fyi @pefernan

### CODE REVIEW

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import org.kie.workbench.common.stunner.core.TestingGraphMockHandler;
+import org.kie.workbench.common.stunner.core.graph.content.definition.Definition;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BPMNProcessVariableDeleteHandlerTest {
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandlerTest.java]: I miss here some test that would assert case, when no node was deleted.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    @Test
+    public void testGetVariableListBusinessRuleTask() {
+        when(definition.getDefinition()).thenReturn(businessRuleTask);
+        Collection<String> delete = bpmnProcessVariableDeleteHandler.delete(graphInstance.graph,
+                                                                            VARIABLE_ID_INPUT);
+        assertTrue(delete.size() == 1);
+    }
+
+    @Test
+    public void testGetVariableListOutputBusinessRule() {
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandlerTest.java]: Just for consistency, it should be named `testGetVariableListOutputBusinessRule` I think.

**Comment by hasys** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+import org.kie.workbench.common.stunner.core.graph.Graph;
+import org.kie.workbench.common.stunner.core.graph.Node;
+import org.kie.workbench.common.stunner.core.graph.content.definition.Definition;
+
+public class BPMNProcessVariableDeleteHandler {
+
+    private final String PROPERTY_IN_PREFIX = "[din]";
+    private final String PROPERTY_OUT_PREFIX = "[dout]";
+
+    public Collection<String> delete(Graph graph,
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java]: Is this `delete` method delete something?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                                               propertyId);
+            if (process) {
+                result.add(nodeVar.getUUID());
+            }
+        });
+
+        return result;
+    }
+
+    private boolean processNode(Node node,
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java]: I am not sure what is processed by this method. Is it checks node for variables? If so, not Only User and BusinessRule tasks have assignments variables. But also all Start events, End events, Sub-Process. Also there are some addition node types will be added soon with Assignment variables. Is it possible to generalize somehow this method?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    private Collection<String> getVariablesName(final String variablesNameString,
+                                                final int startIndex,
+                                                int stopIndex) {
+        Collection<String> properties = new ArrayList();
+
+        if (startIndex >= 0)
+
+        {
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java]: Curly brace is on incorrect place. You can use latest IDEA (or Eclipse) formatter to handle staff like this automatically. https://github.com/kiegroup/droolsjbpm-build-bootstrap/tree/master/ide-configuration

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+            if (stopIndex <= 0) {
+                stopIndex = variablesNameString.length();
+            }
+
+            String name = variablesNameString.substring(startIndex,
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java]: Probably, you are using formatter, but outdated one. Please, import new version and enable `on-commit` automated formatter for your convenience 
image

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java (hunk 4 above)
Comment (jomarko) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java]: +1, please make more clear what is purpose of this method. If it checks variables, I agree with @hasys, it should cover also other process elements.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java (hunk 3 above)
Comment (alepintus) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java]: @hasys Method delete  changed to:  public boolean isVariableBoundToNodes

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java (hunk 4 above)
Comment (alepintus) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/BPMNProcessVariableDeleteHandler.java]: @hasys , @jomarko : After discussing it with @romartin we decided to do it later.

**Comment by romartin** [comment]
ok to test

**Comment by romartin** [comment]
Jenkins please rebuild this

**Comment by romartin** [comment]
Hey @alepintus 
Jenkins is failing due to a failed test for `org.kie.workbench.common.stunner.bpmn.client.forms.fields.variablesEditor.VariablesEditorFieldRendererTest.testRemoveVariable`

This is the error message: 
java.lang.NullPointerException: null
	at org.kie.workbench.common.stunner.bpmn.client.forms.fields.variablesEditor.VariablesEditorFieldRenderer.removeVariable(VariablesEditorFieldRenderer.java:219)
	at org.kie.workbench.common.stunner.bpmn.client.forms.fields.variablesEditor.VariablesEditorFieldRendererTest.testRemoveVariable(VariablesEditorFieldRendererTest.java:89)

Can u please take it a look? Thanks!

**Comment by alepintus** [comment]
Hi @romartin,
Test fixed.

Could you take a look please?

Alessio

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Pull request merged**